### PR TITLE
Add auth parameter for detector and enable auth by default

### PR DESCRIPTION
**Pull request by ignacioli**
#423

### CODE REVIEW

**Comment by ignacioli** [comment]
pulsar_detector image : keep same as charts sn-platform.

**Comment by sijie** [comment]
@ignacioli Can you resolve the conflict?

File: charts/pulsar/templates/detector/pulsar-detector-statefulset.yaml [hunk 1]
```diff
@@ -85,8 +85,19 @@ spec:
         args:
         - >
           bin/pulsar-detector -service-url pulsar://{{ template "pulsar.fullname" . }}-{{ .Values.broker.component }}:{{ .Values.broker.ports.pulsar }};
```
Comment (sijie) [inline charts/pulsar/templates/detector/pulsar-detector-statefulset.yaml:87]: Don't we need to delete this line?

File: charts/pulsar/values.yaml [hunk 2]
```diff
@@ -95,7 +95,7 @@ components:
   # mop
   mop: false
   # pulsar detector
-  pulsar_detector: false
+  pulsar_detector: true
```
Comment (sijie) [inline charts/pulsar/values.yaml:98]: I don't think we should turn this one by default.

File: charts/pulsar/values.yaml [hunk 3]
```diff
@@ -249,7 +249,7 @@ tls:
 # Enable or disable broker authentication and authorization.
 auth:
   authentication:
-    enabled: false
+    enabled: true
```
Comment (sijie) [inline charts/pulsar/values.yaml:252]: Same comment. We shouldn't turn this one by default

**Review by sijie — CHANGES_REQUESTED** [review]

**Pull request closed without merging**